plugin/trino-hive/src/main/java/io/trino/plugin/hive/s3select/S3SelectRecordCursorProvider.java

```diff
@@ -85,21 +81,32 @@ public Optional<ReaderRecordCursorWithProjections> createRecordCursor(
         effectivePredicate = effectivePredicate.filter((column, domain) -> column.isBaseColumn());
 
         String serdeName = getDeserializerClassName(schema);
-        if (CSV_SERDES.contains(serdeName)) {
+        Optional<S3SelectDataType> s3SelectDataTypeOpt = S3SelectSerDeDataTypeMapper.getDataType(serdeName);
```

Comment:
nit: `s3SelectDataTypeOpt` -> `s3SelectDataTypeOptional`

Reply:
Fixed in new commit.